Frameworks/UIKit/UIApplication.mm

```diff
@@ -778,10 +778,22 @@ - (void)sendEvent:(UIEvent*)event {
 }
 
 /**
- @Status Interoperable
+ @Status Caveat
+ @Notes Windows Launcher only supports launching URIs asynchronously. Check if the URI can be opened
+ and return its result to the caller instead of blocking for the URI to be actually opened.
 */
 - (BOOL)openURL:(NSURL*)url {
-    return [_launcher _openURL:url];
+    if ([self canOpenURL:url]) {
+        dispatch_queue_t queue =
+            ([NSThread isMainThread]) ? dispatch_get_main_queue() : dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0);
+        dispatch_async(queue,
+                       ^{
+                           [_launcher _openURL:url];
```

Comment:
It still seems like this check should go into the launcher then.  We have a helper class for URI activation; special handling should go there.

---
In reply to: [72480268](https://github.com/Microsoft/WinObjC/pull/692#discussion_r72480268) [](ancestors = 72480268,72368518)